xunit.runner.visualstudio.desktop/Sinks/VsExecutionSink.cs

```diff
@@ -63,11 +63,11 @@ public override void Dispose()
         TestCase FindTestCase(ITestCase testCase)
         {
             TestCase result;
-            if (testCases.TryGetValue(testCase, out result))
-                return result;
 
-            result = testCases.Where(tc => tc.Key.UniqueID == testCase.UniqueID).Select(kvp => kvp.Value).FirstOrDefault();
```

Comment:
Issue: the test cases here are Proxy objects, comparison operations are costly since they require cross appdomain calls. This function gets called quite frequently (every testcase start, end).

Solution: `ITestCase.UniqueID` identifies a test case, we built a map with `<UniqueID, VSTestCase>`. All the lookup operations are O(1) and do not require cross appdomain calls.